### DISCUSSION TITLE
Yolov4 on MacBook Pro 16-inch using GPU = AMD Radeon Pro 5500M 8 GB

### DISCUSSION THREAD

**Discussion by KohlerVG** [AlexeyAB/darknet#7272] [Q&A]
The title says it, I want to use Yolov4 on macOS using this GPU = AMD Radeon Pro 5500M 8 GB.
Has anyone achieved this?
Thanks,
Kohler

**Comment by AlexeyAB** [comment]
* Try to run your custom `yolov4.cfg / yolov4.weights` without convertation by using OpenCV on MacBook https://docs.opencv.org/master/da/d9d/tutorial_dnn_yolo.html
    * How to install https://docs.opencv.org/master/d0/db2/tutorial_macos_install.html
    * C++: https://github.com/opencv/opencv/blob/master/samples/dnn/object_detection.cpp
    * Python: https://github.com/opencv/opencv/blob/master/samples/dnn/object_detection.py
    
----

* Convert your custom `yolov4.cfg / yolov4.weights` model to ONNX: https://github.com/Tianxiaomo/pytorch-YOLOv4#3-darknet2onnx
    * install OpenMP `brew install libomp`
    * install onnx-runtime `pip install onnxruntime`
    * clone repo `git clone https://github.com/Tianxiaomo/pytorch-YOLOv4`
    * `cd pytorch-YOLOv4`
    * run convertation and demo `python demo_darknet2onnx.py yolov4.cfg yolov4.weights dog.jpg 0`

**Comment by KohlerVG** [comment]
Thank you so much for the instructions! I'll do it and let you know how it goes!
Thanks!
Kohler

**Reply by Dvalin21** [reply]
> Thank you so much for the instructions! I'll do it and let you know how it goes! Thanks! Kohler
> […](#)
> On Jan 22, 2021, at 7:02 PM, Alexey ***@***.***> wrote: Try to run your custom yolov4.cfg / yolov4.weights without convertation by using OpenCV on MacBook https://docs.opencv.org/master/da/d9d/tutorial_dnn_yolo.html <https://docs.opencv.org/master/da/d9d/tutorial_dnn_yolo.html> How to install https://docs.opencv.org/master/d0/db2/tutorial_macos_install.html <https://docs.opencv.org/master/d0/db2/tutorial_macos_install.html> C++: https://github.com/opencv/opencv/blob/master/samples/dnn/object_detection.cpp <https://github.com/opencv/opencv/blob/master/samples/dnn/object_detection.cpp> Python: https://github.com/opencv/opencv/blob/master/samples/dnn/object_detection.py <https://github.com/opencv/opencv/blob/master/samples/dnn/object_detection.py> Convert your custom yolov4.cfg / yolov4.weights model to ONNX: https://github.com/Tianxiaomo/pytorch-YOLOv4#3-darknet2onnx <https://github.com/Tianxiaomo/pytorch-YOLOv4#3-darknet2onnx> install OpenMP brew install libomp install onnx-runtime pip install onnxruntime clone repo git clone https://github.com/Tianxiaomo/pytorch-YOLOv4 cd pytorch-YOLOv4 run convertation and demo python demo_darknet2onnx.py yolov4.cfg yolov4.weights dog.jpg 0 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <[#7272 (comment)](https://github.com/AlexeyAB/darknet/discussions/7272#discussioncomment-303907)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APR34KMNNVRVO54TJVFN2QDS3IN2BANCNFSM4WNTX6IQ>.

Wanted to see if you were able to achieve this?  Thanks.
